cpp/unreal_plugins/SimulationController/Source/SimulationController/SimulationController.cpp

```diff
@@ -90,7 +90,7 @@ void SimulationController::postWorldInitializationEventHandler(UWorld* world, co
 
     ASSERT(world);
 
-    if (world->IsGameWorld() && GEngine->GetWorldContextFromWorld(world)) {
+    if (Config::isEnabled() && world->IsGameWorld() && GEngine->GetWorldContextFromWorld(world)) {
```

Comment:
Better to check in `SimulationController::StartupModule()`. From there, you can prevent everything in `SimulationController` from being executed in a very clean way.

```
void SimulationController::StartupModule()
{
    std::cout << "[SPEAR | SimulationController.cpp] SimulationController::StartupModule" << std::endl;

    ASSERT(FModuleManager::Get().IsModuleLoaded(TEXT("CoreUtils")));
    ASSERT(FModuleManager::Get().IsModuleLoaded(TEXT("OpenBot")));
    ASSERT(FModuleManager::Get().IsModuleLoaded(TEXT("UrdfBot")));

    if (CoreUtils::s_launch_mode_ == LaunchMode::Default) {
        return;
    }
    
    // ...
}

void SimulationController::ShutdownModule()
{
    std::cout << "[SPEAR | SimulationController.cpp] SimulationController::ShutdownModule" << std::endl;

    if (CoreUtils::s_launch_mode_ == LaunchMode::Default) {
        return;
    }
    
    // ...
}
```